### ISSUE TITLE
Html.AjaxGrid question

### ISSUE THREAD

**Issue by Runy89** [NonFactors/MVC5.Grid#193]
Hi. I have a question - in your example - http://mvc-grid.azurewebsites.net/Grid/Ajax - we can see how to create a grid using AjaxGrid helper passing additional html attributes (class, id, data-...), however when we initialize this ajax grid with new MvcGrid() in our JS - we loose all these additional html attributes. Is there a way to pass them that i did not find in examples or maybe you could add functionality to keep those additional html attributes? Thanks in advance.

**Comment by Muchiachio** [comment]
Ajax helper attributes should only be used for temporary initialization stuff like id.
If you are fine with using attributes after grid initialization, you can just add them in the grid partial, using one of the extensions like `.Attributed` or `.RowAttributed`, otherwise you will need to duplicate in ajax helper and the grid.

**Comment by Runy89** [comment]
OK, but .Attributed & .RowAttributed would add html attributes to the <table> element inside the .mvc-grid div -  which AjaxGrid helper would add them to. So different elements would end up with attributes. If this behaviour is correct and there is no need to add support for AjaxGrid helper html attributes, then it's ok and i can work my ways around it, but if you'd think cases and scenarios where this feature might be useful - it'd be great.

**Comment by Muchiachio** [comment]
Let me rephrase, ajax helper is only to set id, which you can use to initialize the grid.
Everything else and the same container id can be set through extensions I mentioned.

I left it that way, because it's just a container for initialization and I didn't want to add even more duplication between initialization and the partial view. I'm also not sure about the scenarios in which it would be needed over normal table attributes. Maybe you have some scenarios?

**Comment by Runy89** [comment]
well my current sistuation is that i have multiple grids on page they differ by behaviour - some have additional custom filters, other - do not have that option. for them to work properly i need additional classes and data attributes, since id would be too unique for common approach. and that is how i stumbled upon this question. i can wrap them in additional divs and that would fix that, but since table itseld is already wrapped with div and i can pass html attributes there - everything was smooth and clean until i tried ajax approach.

**Comment by Muchiachio** [comment]
It looks like single time initialization for custom filters, so you can use temporary attributes.
Because filtering options are preserved after ajax reload anyway.

**Comment by Runy89** [comment]
Yes. I'm still on my way to find the best solution here. Thank you for all your time and answers.
